Node/core/lib/dialogs/EntityRecognizer.js

```diff
@@ -176,13 +176,34 @@ var EntityRecognizer = (function () {
                 score = Math.min(0.5 + (value.length / utterance.length), 0.9);
             }
             else {
-                var matched = '';
+                var matched = {};
                 tokens.forEach(function (token) {
                     if (value.indexOf(token) >= 0) {
-                        matched += token;
+                        if (!matched[token]) {
+                            matched[token] = 1;
+                        }
                     }
                 });
-                score = matched.length / value.length;
```

Comment:
The problem from calculating the score this way is that on [line 181][L181] it will add words that shouldn't be added to `matched`. 

In #3516 `EntityRecognizer.findAllMatches()` added `"a"` and `"in"` to `matched`, because these two words can be found in `"email"` and `"link"`.  Thus `matched` would equal `"alinkinemail"`. The entity value the user was looking for was `"email link"` and when `matched.length` is divided by `value.length`, the returned score is 1.2.


  [L181]: https://github.com/Microsoft/BotBuilder/pull/3702/files#diff-a5cbf2306473b89ce8f6eb849e7f09c5L181